wasm-node/javascript/src/index-deno.ts

```diff
@@ -161,6 +161,10 @@ function connect(config: ConnectionConfig, forbidTcp: boolean, forbidWs: boolean
             inner: Deno.connect({
                 hostname: tcpParsed[2],
                 port: parseInt(tcpParsed[3]!, 10),
+            }).catch((error) => {
+                socket.destroyed = true;
+                config.onConnectionReset(error.toString());
+                return null!;
```

Comment:
Could you maybe move this as the second parameter of the `then` below?
This would guarantee that the first closure of the `then` never gets a `null`

Also:

```suggestion
            }).catch((error) => {
                socket.destroyed = true;
                config.onConnectionReset(error.toString());
                return null;
```

Reply:
That first closure (and `.send` too) checks `socket.destroyed`, so doesn't use `null`.

Either `null!` is required, or change of type to `TcpConnection { socket(inner: Promise<Deno.TcpConn | null>) }`

Reply:
> or change of type to TcpConnection { socket(inner: Promise<Deno.TcpConn | null>) }

Well, yes, that option.
As far as I understand, `null!` lies to the type checker. It's an escape hatch for situations where TypeScript isn't good enough, and not something you're supposed to use normally.